pkg/operator/k8sutil/node.go

```diff
@@ -279,3 +280,74 @@ func NodeIsInRookNodeList(targetNodeName string, rookNodes []rookalpha.Node) boo
 	}
 	return false
 }
+
+// ValidateLabelKey validates Key of the Label against regex
+func ValidateLabelKey(s string) bool {
+	re := regexp.MustCompile(`^(((([a-zA-Z0-9]+[.]){1,}[a-zA-Z0-9]+){1}[\/]{1})?)([a-zA-Z0-9](([a-zA-Z0-9]*[\-_.][a-zA-Z0-9]+){1,})[a-zA-Z0-9]$)`)
+	return re.MatchString(s)
+}
+
+// ValidateLabelValue validates Value of the Label against regex
+func ValidateLabelValue(s string) bool {
+	re := regexp.MustCompile(`(?m)^[a-z0-9A-Z]+([a-z0-9A-Z]+[\-._])*[a-z0-9A-Z]+$`)
+	return re.MatchString(s)
+}
+
+// AddNodeAffinity will return v1.NodeAffinity or error
+func AddNodeAffinity(nodeAffinity string) (*v1.NodeAffinity, error) {
+	newNodeAffinity := &v1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+			NodeSelectorTerms: []v1.NodeSelectorTerm{
+				{},
+			},
+		},
+	}
+	nodeLabels := strings.Split(nodeAffinity, ";")
+	// For each label in 'nodeLabels', retrieve (key,value) pair and create nodeAffinity
+	// '=' separates key from values
+	// ',' separates values
+	for _, nodeLabel := range nodeLabels {
+		// If tmpNodeLabel is an array of length > 1
+		// [0] is Key and [1] is comma separated values
+		tmpNodeLabel := strings.Split(nodeLabel, "=")
+		if len(tmpNodeLabel) > 1 {
+			nodeLabelKey := tmpNodeLabel[0]
+			tmpNodeLabelValue := tmpNodeLabel[1]
+			nodeLabelValues := strings.Split(tmpNodeLabelValue, ",")
+			if nodeLabelKey != "" && len(nodeLabelValues) > 0 {
+				re := ValidateLabelKey(nodeLabelKey)
+				if !re {
+					return nil, fmt.Errorf("Invalid Label Key")
```

Comment:
Perhaps we could print the value so that will ease the debugging?

Reply:
In addition to printing the value that is invalid, please keep the errors in lowercase as the other errors returned by Rook.

Same goes for the other error returns.

Reply:
Done.